lib/kubeclient/informer.rb

```diff
@@ -1,7 +1,7 @@
 module Kubeclient
   # caches results for multiple consumers to share and keeps them updated with a watch
   class Informer
-    def initialize(client, resource_name, reconcile_timeout: 15 * 60, logger: nil)
+    def initialize(client, resource_name, options: {}, reconcile_timeout: 15 * 60, logger: nil)
```

Comment:
What do you think about splitting this to 2 options? It seems that the `watch` and the `get` APIs accept different options

Reply:
Seems overkill. I would leave it this way until there is strong use case for different options. Options that control the result of the requests are same.

Reply:
A good question, let's see...  
#### Shared options:

- :namespace (string)
- :label_selector (string)
- :field_selector (string)
- :resource_version (string) — but informer overrides this in both get & watch
- :as — informer might break if you set this in `options`, or if you've customized `client`'s `@as`?  Probably informer SHOULD override this?  However should do `format_response` processing in `@watching.each { |q| q << notice }`?  Or should it be an error?  => You know what, I'll offer a separate issue about this.

#### Get-only options:

- :limit (integer)
- :continue (string) — these two are maybe worth using inside `fill_cache` (to limit apiserver RAM usage on huge collections), but make no sense to set externally.

#### Watch-only options:

- :allow_watch_bookmarks (bool) — Current informer code would log "Unsupported event type" but wouldn't benefit.  Would be nice to use internally by informer, but if we do that I think it can be unconditional, makes no sense to set externally IMHO.
- :name (string) — Informer does not support this currently, would need to switch to `get_entity`.  But that's implementation detail; if we support this, I wouldn't want caller to have to pass name in two different ways, a shared `options` interface with special treatment for `:name` remains good.

=> So I think @DocX is right here.